train_concept_identification.py

```diff
@@ -41,6 +42,94 @@ def compute_loss(criterion, logits, gold_outputs):
   loss = criterion(flattened_logits, flattened_gold_outputs)
   return loss
 
+
+def compute_fScore(gold_outputs,
+                   predicted_outputs,
+                   vocabs: Vocabs):
+    """Computes f_score, precision, recall.
+
+  Args:
+    gold_outputs: Gold outputs, shape (output seq len, batch size)
+    predicted_outputs: Predicted outputs, shape (output seq len, batch size)
+
+  Returns:
+    f_score
+  """
+
+    predicted_outputs_no_padding, gold_outputs_no_padding = tensor_to_list(gold_outputs, predicted_outputs)
+
+    ids_to_concepts_list_gold = list(vocabs.concept_vocab.keys())
+    concepts_as_list_gold = []
+    #TODO store the gold data before numericalization and use it here
+    for sentence in gold_outputs_no_padding:
+        for id in sentence:
+            if ids_to_concepts_list_gold[id] != UNK:
+                concepts_as_list_gold.append(ids_to_concepts_list_gold[id])
+
+
+    ids_to_concepts_list_predicted = list(vocabs.concept_vocab.keys())
+    concepts_as_list_predicted = []
+    for sentence in predicted_outputs_no_padding:
+        for id in sentence:
+            if ids_to_concepts_list_predicted[int(id)] != UNK:
+                concepts_as_list_predicted.append(ids_to_concepts_list_predicted[int(id)])
```

Comment:
This code is a repetition of the code a few lines above. Please extract in a function
It can be named remove_unk / remove_unk_from_sequence or something along those lines